packages/react-magma-dom/src/components/Hyperlink/index.tsx

```diff
@@ -73,16 +78,20 @@ export const Hyperlink = React.forwardRef<HTMLAnchorElement, HyperlinkProps>(
         </ClassNames>
       );
     } else {
+      const LinkStyledAsButton = BaseStyledButton.withComponent('a');
+      const HyperlinkComponent =
+        styledAs === 'Button' ? LinkStyledAsButton : StyledLink;
```

Comment:
In a future PR, we should make this an enum to be consistent with all the other constrained props.